src/python/TaskWorker/Actions/RetryJob.py

```diff
@@ -380,7 +383,48 @@ def check_exit_code(self):
         return 0
 
     ##= = = = = RetryJob = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
- 
+
+    def check_corrupted_file(self, exitCode):
+        corruptedFile = False
+        fname = os.path.realpath("WEB_DIR/job_out.%s.%d.txt" % (self.job_id, self.crab_retry))
+        self.logger.debug(f'exit code {exitCode}, look for corrupted file in {fname}')
+        with open(fname, encoding='utf-8') as fd:
+            for line in fd:
+                if line.startswith("== CMSSW:") and "Fatal Root Error:" in line:
+                    corruptedFile = True
+                    self.logger.info("Corrupted input file found")
+                    self.logger.debug(line)
+                    errorLines = [line]
+                    # file name is n next line
+                    continue
+                if corruptedFile:
+                    errorLines.append(line)
+                    # extract the '/store/...root' part of this line
+                    fragment1 = line.split('/store/')[1]
+                    fragment2 = fragment1.split('.root')[0]
+                    inputFileName = f"/store/{fragment2}.root"
+                    RSE = self.site
+                    RSE = RSE if not RSE.startswith('T1') else f'{RSE}_Disk'
+                    self.logger.info(f"RSE: {RSE} - ec: {exitCode} - file: {inputFileName}")
+                    break
+        if corruptedFile:
+            # note it  down
+            reportFileName = f'{self.reqname}.job.{self.job_id}.{self.crab_retry}.json'
+            corruptionMessage = {'DID': f'cms:{inputFileName}', 'RSE': RSE,
+                                  'exitCode': exitCode, 'message': errorLines}
+            with open(reportFileName, 'w', encoding='utf-8') as fp:
+                json.dump(corruptionMessage, fp)
+            self.logger.info('corruption message prepared, gfal-copy to EOS')
+            proxy = os.getenv('X509_USER_PROXY')
+            self.logger.info(f"X509_USER_PROXY = {proxy}")
+            reportLocation = 'gsiftp://eoscmsftp.cern.ch/eos/cms/store/temp/user/corrupted/'
```

Comment:
Does EOS has any problem when we store too many files single directory? We have 100k-200k FileOpen error jobs for 7 days period https://monit-grafana.cern.ch/goto/jHHYEMiIz?orgId=11
And how do we plan to clean it up?

Reply:
my current plan is:
 - write reports in `..../corrupted/new`
 -  have a crontab which processes them, marks rucio replica as suspicious (or bad) as proper and move to `.../corrupted/done` )and cleans done after e.g. a couple weeks)
 - leave it to later to decide if need to structure directory by day or similar. But I currently expect that only a very small fraction of file open errors have a "Fatal Root Error".